vector/src/main/java/im/vector/riotx/features/reactions/EmojiSearchResultController.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.riotx.features.reactions
+
+import android.graphics.Typeface
+import com.airbnb.epoxy.TypedEpoxyController
+import im.vector.riotx.EmojiCompatFontProvider
+import im.vector.riotx.R
+import im.vector.riotx.core.resources.StringProvider
+import im.vector.riotx.core.ui.list.genericFooterItem
+import javax.inject.Inject
+
+class EmojiSearchResultController @Inject constructor(val stringProvider: StringProvider,
+                                                      fontProvider: EmojiCompatFontProvider)
+    : TypedEpoxyController<EmojiSearchResultViewState>() {
+
+    var emojiTypeface: Typeface? = fontProvider.typeface
+
+    init {
+        fontProvider.addListener(object : EmojiCompatFontProvider.FontProviderListener {
```

Comment:
You add listeners you never remove on a singleton instance :/

Reply:
done